vg/font.go

```diff
@@ -149,42 +159,73 @@ type FontExtents struct {
 
 // Extents returns the FontExtents for a font.
 func (f *Font) Extents() FontExtents {
-	bounds := f.font.Bounds(fixed.Int26_6(f.Font().FUnitsPerEm()))
-	scale := f.Size / Points(float64(f.Font().FUnitsPerEm()))
+	var (
+		// TODO(sbinet): re-use a Font-level sfnt.Buffer instead?
+		buf  sfnt.Buffer
+		ppem = fixed.Int26_6(f.font.UnitsPerEm())
+	)
+
+	met, err := f.font.Metrics(&buf, ppem, f.hinting)
+	if err != nil {
+		panic(fmt.Errorf("could not extract font extents: %v", err))
+	}
+	scale := f.Size / Points(float64(ppem))
 	return FontExtents{
-		Ascent:  Points(float64(bounds.Max.Y)) * scale,
-		Descent: Points(float64(bounds.Min.Y)) * scale,
-		Height:  Points(float64(bounds.Max.Y-bounds.Min.Y)) * scale,
+		Ascent:  Points(float64(met.Ascent)) * scale,
+		Descent: Points(float64(-met.Descent)) * scale, // sfnt-descent is >0
+		Height:  Points(float64(met.Height)) * scale,
 	}
 }
 
 // Width returns width of a string when drawn using the font.
 func (f *Font) Width(s string) Length {
-	// scale converts truetype.FUnit to float64
-	scale := f.Size / Points(float64(f.font.FUnitsPerEm()))
+	var (
+		ppem = fixed.Int26_6(f.font.UnitsPerEm())
```

Comment:
This name lives long enough that it probably wants a longer name, `pixelsPerEm`?